storm-core/src/jvm/backtype/storm/scheduler/resource/User.java

```diff
@@ -0,0 +1,346 @@
+        }
+        return sum;
+    }
+
+    public double getMemoryResourceUsedByUser() {
+        double sum = 0.0;
+        for (TopologyDetails topo : this.runningQueue) {
+            sum += topo.getTotalRequestedMemOnHeap() + topo.getTotalRequestedMemOffHeap();
+        }
+        return sum;
+    }
+
+    public Double getMemoryResourceGuaranteed() {
+        return this.resourcePool.get("memory");
+    }
+
+    public Double getCPUResourceGuaranteed() {
+        return this.resourcePool.get("cpu");
+    }
+
+    public TopologyDetails getNextTopologyToSchedule() {
+        for (TopologyDetails topo : this.pendingQueue) {
+            if (!this.attemptedQueue.contains(topo)) {
+                return topo;
+            }
+        }
+        return null;
+    }
+
+    public boolean hasTopologyNeedSchedule() {
+        return (!this.pendingQueue.isEmpty());
+    }
+
+    public TopologyDetails getRunningTopologyWithLowestPriority() {
+        if (this.runningQueue.isEmpty()) {
+            return null;
+        }
+        return this.runningQueue.last();
+    }
+
+    @Override
+    public int hashCode() {
+        return this.userId.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (!(o instanceof User)) {
+            return false;
+        }
+        return this.getId().equals(((User) o).getId());
+    }
+
+    @Override
+    public String toString() {
+        return this.userId;
+    }
+
+    public String getDetailedInfo() {
+        String ret = "\nUser: " + this.userId;
+        ret += "\n - " + " Resource Pool: " + this.resourcePool;
+        ret += "\n - " + " Running Queue: " + this.runningQueue + " size: " + this.runningQueue.size();
+        ret += "\n - " + " Pending Queue: " + this.pendingQueue + " size: " + this.pendingQueue.size();
+        ret += "\n - " + " Attempted Queue: " + this.attemptedQueue + " size: " + this.attemptedQueue.size();
+        ret += "\n - " + " Invalid Queue: " + this.invalidQueue + " size: " + this.invalidQueue.size();
+        ret += "\n - " + " CPU Used: " + this.getCPUResourceUsedByUser() + " CPU guaranteed: " + this.getCPUResourceGuaranteed();
+        ret += "\n - " + " Memory Used: " + this.getMemoryResourceUsedByUser() + " Memory guaranteed: " + this.getMemoryResourceGuaranteed();
+        ret += "\n - " + " % Resource Guarantee Used: \n -- CPU: " + this.getCPUResourcePoolUtilization()
+                + " Memory: " + this.getMemoryResourcePoolUtilization() + " Average: " + this.getResourcePoolAverageUtilization();
+        return ret;
+    }
+
+    public static String getResourcePoolAverageUtilizationForUsers(Collection<User> users) {
+        String ret = "";
+        for (User user : users) {
+            ret += user.getId() + " - " + user.getResourcePoolAverageUtilization() + " ";
+        }
+        return ret;
+    }
+
+    /**
+     * Comparator that sorts topologies by priority and then by submission time
```

Comment:
maybe a short comment saying that ties go to the topology that has been running longer